libp2p/connmanager.nim

```diff
@@ -236,12 +237,17 @@ proc cleanupConn(c: ConnManager, conn: Connection) {.async.} =
 
   trace "Connection cleaned up", conn
 
-proc peerStartup(c: ConnManager, conn: Connection) {.async.} =
+proc onConnUpgraded(c: ConnManager, conn: Connection) {.async.} =
   try:
     trace "Triggering connect events", conn
+    # NOTE: make sure the upgrade event
+    # happens *before* any other events
+    # are triggered
```

Comment:
why is this important? ie anything that's waiting on this future may run either before or after the peer event trigger below